include/fiddle/interaction/interaction_utilities.h

```diff
@@ -238,5 +239,31 @@ namespace fdl
                        const Vector<double>         &spread_values);
 
 
+  /**
+   * Intersec line with edge .
+   * WARNING: This code is specialized to the case in which q is a unit vector
+ aligned with the coordinate axes.
+   */
+  bool
+  intersect_line_with_edge(std::vector<std::pair<double, Point<1>> >& t_vals,
+                           const DoFHandler<1, 2>::active_cell_iterator elem,
+                           const dealii::MappingQ<1, 2> &mapping,
+                           dealii::Point<2> r,
+                           dealii::Tensor<1,2> q,
+                           const double tol = 0.0);
```

Comment:
While we may want to write two versions of this function for spacedim = 2 and spacedim = 3, that's generally an implementation detail - the general intersection methods should be templated on the dimension.

Other comments
- If there are preconditions then those should be assertions, not warnings listed in the documentation.
- Please use full words with underscores (or common abbreviations like `tria`) for input arguments
- Add doxygen-style documentation (like the other functions in this header)
- We should be using `Mapping` and not `MappingQ` (i.e., this won't work with simplices, and it definitely won't work with Lagrangian deformations via `MappingFEField`)

Reply:
Thank you David. For the last comment, do you have any example or something similar related to judging what type of cell (whether it's simplices or quad) and type of mapping ?

Reply:
I recommend assuming a linear mapping for now - we can add some assertions and support for multiple intersections later. If you need the element reference cell you can always do `cell->reference_cell()` to distinguish between `Quadrilateral` and `Triangle` when `dim = 2` and `spacedim = 3`, though we shouldn't worry about quads yet.

Bear in mind that the position of the element is not defined by `cell->vertex(0)` etc: that's the reference configuration. The actual position is given by the mapping which uses a finite element field (`MappingFEField`) to compute the absolute positions.

Reply:
When we detect intersections we will need to read back Eulerian data - hence we should be using SAMRAI data here (in particular something like `SideIndex`) so that we don't lose information.